### PR TITLE
 DCM:28: Map indicator mappings using index in Run Report view

### PR DESCRIPTION
**Pull request by Bawanthathilan**
## The issue I worked on

See https://issues.openmrs.org/browse/DCM-28
## Description of what I changed:
- Replace the values of the mappings selector's options with mapping index
- Introduce a global variable to keep the selected mapping
- Replace existing period type extraction logic with the introduced selectedMapping variable.

### CODE REVIEW

File: omod/src/main/webapp/resources/dhisconnector-runreports.js [hunk 1]
```diff
@@ -124,6 +119,27 @@ function toggleCustomRangeCheckbox(checkDisable) {
     }
 }
 
+function getPeriodType(){
```
Comment (jayasanka-sack) [inline omod/src/main/webapp/resources/dhisconnector-runreports.js:122]: Where this function came from?

File: omod/src/main/webapp/resources/dhisconnector-runreports.js [hunk 2]
```diff
@@ -97,14 +91,15 @@ function onMappingSelect() {
...
         });
-    } else if (selectedPeriodType === 'Monthly') {
+    } else if (selectedMapping.periodType === 'Monthly') {
         // set up monthly
         jQuery('#periodSelector').monthpicker({
             pattern: 'yyyymm'
         });
-    } else {
+    } else
+    {
```
Comment (jayasanka-sack) [inline omod/src/main/webapp/resources/dhisconnector-runreports.js:100]:
```suggestion
    } else {
```

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Good job Bawantha!

I added a single comment. Could you please check it?

File: omod/src/main/webapp/resources/dhisconnector-runreports.js (hunk 1 above)
Comment (Bawanthathilan) [reply omod/src/main/webapp/resources/dhisconnector-runreports.js]: ill fix it @jayasanka-sack

Comment (Bawanthathilan) [reply omod/src/main/webapp/resources/dhisconnector-runreports.js]: please check now

File: omod/src/main/webapp/resources/dhisconnector-runreports.js [hunk 3]
```diff
@@ -430,11 +409,12 @@ function createDownload(content, contentType, extension) {
 function generateDXFDownload() {
     if (validateForm()) {
         buildDXFJSON().then(function () {
-        	createDownload(JSON.stringify(dxfJSON), 'application/json', '.dxf.json');
+            createDownload(JSON.stringify(dxfJSON), 'application/json', '.dxf.json');
         });
     }
 }
 
+
```
Comment (jayasanka-sack) [inline omod/src/main/webapp/resources/dhisconnector-runreports.js:417]: Remove this too

Comment (Bawanthathilan) [reply omod/src/main/webapp/resources/dhisconnector-runreports.js]: please check now

**Review by jayasanka-sack — APPROVED** [review]
Thanks for getting this done, @Bawanthathilan!

**Comment by dkayiwa** [comment]
@jayasanka-sack did you forget to update the ticket? And the fix version?

**Pull request merged**